File: servers/MasterServices/Models/ChatSession.cs
using System.ComponentModel.DataAnnotations;

namespace MasterServices.Models
{
    public class ChatSession
    {
        [Key]
        public string Id { get; set; } = string.Empty;
        
        [Required]
        public string SessionId { get; set; } = string.Empty; // n8n session id
        
        [Required]
        public string Title { get; set; } = string.Empty;
        
        public Result? Result { get; set; }
        
        public string? XrayImageUrl { get; set; }
        
        public List<ChatItem>? ChatItems { get; set; }
        
        public List<Report>? Reports { get; set; }
        
        [Required]
        public bool IsDeleted { get; set; }
        
        public DateTime? CreatedDate { get; set; }
        
        public DateTime? UpdatedDate { get; set; }
    }

    public class Result
    {
        public List<DiseasePrediction>? PredictedDiseases { get; set; }
        
        public List<DiseasePrediction>? Top5Diseases { get; set; }
        
        public GradcamAnalyses? GradcamAnalyses { get; set; } // s3 urls
        
        public string? AttentionMap { get; set; } // s3 url
        
        public IndividualAnalyses? IndividualAnalyses { get; set; }
        
        public string? ConciseConclusion { get; set; }
        
        public string? ComprehensiveAnalysis { get; set; }
    }

    public class DiseasePrediction
    {
        [Required]
        public string Disease { get; set; } = string.Empty;
        
        [Required]
        public double Confidence { get; set; }
    }

    public class GradcamAnalyses
    {
        public string? Top1Pneumonia { get; set; }
        
        public string? Top2Consolidation { get; set; }
        
        public string? Top3Effusion { get; set; }
        
        public string? Top4Atelectasis { get; set; }
        
        public string? Top5Cardiomegaly { get; set; }
    }

    public class IndividualAnalyses
    {
        public string? Top1Pneumonia { get; set; }
        
        public string? Top2Consolidation { get; set; }
        
        public string? Top3Effusion { get; set; }
        
        public string? Top4Atelectasis { get; set; }
        
        public string? Top5Cardiomegaly { get; set; }
    }

    public class ChatItem
    {
        [Required]
        public string Content { get; set; } = string.Empty;
        
        public List<string>? ImageUrls { get; set; }
        
        public bool? IsBot { get; set; }
        
        public DateTime? CreatedDate { get; set; }
        
        public ChatItemMetaData? MetaData { get; set; }
    }

    public class ChatItemMetaData
    {
        public string? PubmedQueryUrl { get; set; }
        
        public List<string>? PubmedFetchUrl { get; set; }
    }
}

File: web_app/types/chatsession.ts

export type ChatSessionRequest = {
    title: string;
    xrayImage: Buffer;
}


export type ChatSession = {
    id: string;
    title: string;
    result?: Result;
    xrayImageUrl?: string;
    chatItems?: ChatItem[];
    reports?: Report[];
    isDeleted: boolean;
    createdDate?: string;
    updatedDate?: string;
}

export type Result = {
    predictedDiseases: DiseasePrediction[];
    top5Diseases: DiseasePrediction[];
    gradcamAnalyses: { // s3 urls - dynamic keys from backend (e.g., "top1_Hernia", "top2_Cardiomegaly")
        [key: string]: string;
    };
    attentionMap: string; // s3 url
    individualAnalyses: {
        top1_Pneumothorax: string;
        top2_Atelectasis: string;
        top3_Edema: string;
        top4_Pneumonia: string;
        top5_Pleural_Thickening: string;
        [key: string]: string; // Allow additional dynamic keys
    };
    conciseConclusion: string;
    comprehensiveAnalysis: string;
}

export type DiseasePrediction = {
    disease: string;
    confidence: number;
}

export type ChatItem = {
    content: string;
    imageUrls?: string[];
    isBot?: boolean;
    createdDate?: string;
    metaData?: {
        pubmedQueryUrl?: string;
        pubmedFetchUrl?: string[];
    };
}

export type Report = {
    id?: string;
    content?: string;
    createdDate?: string;
}